pycaret/classification.py

```diff
@@ -599,17 +599,32 @@ def setup(
 
     logger.info("Checking Exceptions")
 
-    # checking data type
-    if hasattr(data, "shape") is False:
+    # Checking data type
+    if not isinstance(data, pd.DataFrame):
         raise TypeError("data passed must be of type pandas.DataFrame")
 
-    # checking train size parameter
-    if type(train_size) is not float:
-        raise TypeError("train_size parameter only accepts float value.")
-
-    # checking sampling parameter
-    if type(sampling) is not bool:
-        raise TypeError("sampling parameter only accepts True or False.")
+    # Checking float parameters in range 0-1
+    floats_in_0_1 = ['train_size', 'rare_level_threshold', 'outliers_threshold',
+                     'multicollinearity_threshold', 'polynomial_threshold',
+                     'feature_selection_threshold', 'interaction_threshold']
+    
+    for param in floats_in_0_1:
+        if not isinstance(eval(param), float) or not 0 < eval(param) < 1:
```

Comment:
Is it a good idea to use `eval` here? It's not a web app, so it's not probably a huge security risk, but it doesn't sit well with me. You could just do `locals()['param']` instead. `locals()` and `globals()` return dictionaries of currently set local and global variables in the form `name : value`.